primitives/consensus/common/src/import_queue.rs

```diff
@@ -15,272 +15,5 @@
-	let parent_hash = header.parent_hash().clone();
-
-	let import_handler = |import| {
-		match import {
-			Ok(ImportResult::AlreadyInChain) => {
-				trace!(target: "sync", "Block already in chain {}: {:?}", number, hash);
-				Ok(BlockImportResult::ImportedKnown(number, peer.clone()))
-			},
-			Ok(ImportResult::Imported(aux)) => Ok(BlockImportResult::ImportedUnknown(number, aux, peer.clone())),
-			Ok(ImportResult::MissingState) => {
-				debug!(target: "sync", "Parent state is missing for {}: {:?}, parent: {:?}", number, hash, parent_hash);
-				Err(BlockImportError::MissingState)
-			},
-			Ok(ImportResult::UnknownParent) => {
-				debug!(target: "sync", "Block with unknown parent {}: {:?}, parent: {:?}", number, hash, parent_hash);
-				Err(BlockImportError::UnknownParent)
-			},
-			Ok(ImportResult::KnownBad) => {
-				debug!(target: "sync", "Peer gave us a bad block {}: {:?}", number, hash);
-				Err(BlockImportError::BadBlock(peer.clone()))
-			},
-			Err(e) => {
-				debug!(target: "sync", "Error importing block {}: {:?}: {:?}", number, hash, e);
-				Err(BlockImportError::Other(e))
-			}
-		}
-	};
-
-	match import_handler(import_handle.check_block(BlockCheckParams {
-		hash,
-		number,
-		parent_hash,
-		allow_missing_state: block.allow_missing_state,
-		import_existing: block.import_existing,
-	}).await)? {
-		BlockImportResult::ImportedUnknown { .. } => (),
-		r => return Ok(r), // Any other successful result means that the block is already imported.
-	}
-
-	let started = wasm_timer::Instant::now();
-	let (mut import_block, maybe_keys) = verifier.verify(
-		block_origin,
-		header,
-		justifications,
-		block.body
-	).await.map_err(|msg| {
-		if let Some(ref peer) = peer {
-			trace!(target: "sync", "Verifying {}({}) from {} failed: {}", number, hash, peer, msg);
-		} else {
-			trace!(target: "sync", "Verifying {}({}) failed: {}", number, hash, msg);
-		}
-		if let Some(metrics) = metrics.as_ref() {
-			metrics.report_verification(false, started.elapsed());
-		}
-		BlockImportError::VerificationFailed(peer.clone(), msg)
-	})?;
-
-	if let Some(metrics) = metrics.as_ref() {
-		metrics.report_verification(true, started.elapsed());
-	}
-
-	let mut cache = HashMap::new();
-	if let Some(keys) = maybe_keys {
-		cache.extend(keys.into_iter());
-	}
-	import_block.import_existing = block.import_existing;
-	let mut import_block = import_block.clear_storage_changes_and_mutate();
-	if let Some(state) = block.state {
-		import_block.state_action = StateAction::ApplyChanges(crate::StorageChanges::Import(state));
-	} else if block.skip_execution {
-		import_block.state_action = StateAction::Skip;
-	} else if block.allow_missing_state {
-		import_block.state_action = StateAction::ExecuteIfPossible;
-	}
-
-	let imported = import_handle.import_block(import_block, cache).await;
-	if let Some(metrics) = metrics.as_ref() {
-		metrics.report_verification_and_import(started.elapsed());
-	}
-	import_handler(imported)
-}
+pub type CacheKeyId = [u8; 4];
```

Comment:
Let's move this to `block_import.rs` and remove this file.

Reply:
Happy to do this but it's referenced by some other (client only) bits in primitives that I have to move out first so will do that in a separate PR.